### PR TITLE
try starting retriever when 80% of accounts are ready

### PR DESCRIPTION
**Pull request by albertcui**

### CODE REVIEW

File: retriever.js [hunk 1]
```diff
@@ -100,14 +107,23 @@ async.each(a, function(i, cb) {
...
 
     function allDone() {
-        if (dotaReady && relationshipReady && !launched) {
+        if (dotaReady && relationshipReady) {
             count += 1;
-            console.log("acct %s ready, %s/%s", i, count, users.length);
+            console.log("acct %s ready, %s/%s", i, count, users.length);    
+        }
+        
+        if (!launched) {
```
Comment (howardchung) [inline retriever.js]: I think you need to check all three conditions before cb()

Comment (howardchung) [inline retriever.js]: Also this will prevent the last 20% of accounts from cb()?

Comment (albertcui) [inline retriever.js]: The last 20% of accounts will cb() from the top if statement.

Comment (howardchung) [inline retriever.js]: i guess.  I feel that this is kind of messy.  Maybe we should make logging in and launching separate things

File: retriever.js [hunk 2]
```diff
@@ -12,13 +12,68 @@ var steamObj = {};
...
+        error: err
+    });
+});
+
+var server = app.listen(port, function() {
+    var host = server.address().address;
+    console.log('[RETRIEVER] listening at http://%s:%s', host, port);
+});
+    
 async.each(a, function(i, cb) {
```
Comment (howardchung) [inline retriever.js]: does this need to be async if you don't have a callback?  Is the callback required if you're using async?

Comment (albertcui) [inline retriever.js]: It is not required, and async is better :P

Comment (howardchung) [inline retriever.js]: how is it any better?

**Pull request merged**